backend/svirt.pm

```diff
@@ -226,6 +249,36 @@ sub start_serial_grab {
     }
 }
 
+# opens another SSH connection to grab the serial console with the specified port
+sub open_serial_console_via_ssh {
+    my ($self, $name, $port) = @_;
+
+    # FIXME: don't hardcode path
+    my $read_serial_script = Mojo::File->new('/hdd/openqa-devel/repos/os-autoinst/read_serial_dev_from_virsh_xml.pm')->slurp;
+    $read_serial_script =~ s/\\/\\\\/g;
+    $read_serial_script =~ s/\"/\\\"/g;
+    $read_serial_script =~ s/\$/\\\$/g;
+    $read_serial_script =~ s/\</\\\</g;
+    $read_serial_script =~ s/\>/\\\>/g;
+
+    bmwqemu::diag("Starting SSH connection to connect to libvirt domain $name via serial port $port");
+    my $credentials = $self->read_credentials_from_virsh_variables;
+    my $ssh = $self->new_ssh_connection(%$credentials);
+    my $chan = $ssh->channel();
+    die 'No channel found' unless $chan;
+    $chan->blocking(0);
+    $chan->pty(1);
+    my $command = "screen \"\$(virsh dumpxml \"$name\" | perl -e \"$read_serial_script\" \"$port\")\"";
```

Comment:
I think `socat` is more appropriate here than screen. We just want the bytes to be transferred from the serial device/socket to the SSH socket.

Reply:
Infact in QEMU you can expose the serial device as a socket which is accessible directly from the network. I wonder if this is also possible here?

Reply:
I didn't know that tool. I'm going to read the man page. Since this runs on the svirt host, we might need to install it there first. This leads to the question: How are dependencies on those "jump-hosts" handled?

Reply:
I changed this now to use `virsh console`.